upup/pkg/fi/cloudup/awstasks/dnszone.go

```diff
@@ -132,7 +132,7 @@ func (e *DNSZone) findExisting(cloud awsup.AWSCloud) (*route53.GetHostedZoneOutp
 
 	var zones []*route53.HostedZone
 	for _, zone := range response.HostedZones {
-		if aws.StringValue(zone.Name) == findName {
+		if aws.StringValue(zone.Name) == findName && *zone.Config.PrivateZone == *e.Private {
```

Comment:
FYI, it is often safer to use `aws.BoolValue` rather than risk a nil-pointer dereference.  But I'm guessing they are both always non-nil..